### PR TITLE
[scheduler] Fix xcode not being forwarded to recipe module

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/95429

This was an issue with platform specific xcode not being applied to the properties

### CODE REVIEW

File: app_dart/lib/src/model/ci_yaml/target.dart [hunk 1]
```diff
@@ -70,9 +70,9 @@ class Target {
     mergedProperties['dependencies'] = mergedDependencies.values.map((Dependency dep) => dep.toJson()).toList();
 
     // xcode is a special property as there's different download policies if its in the devicelab.
-    if (properties.containsKey('xcode')) {
+    if (mergedProperties.containsKey('xcode')) {
       final Object xcodeVersion = <String, Object>{
-        'sdk_version': properties['xcode']!,
+        'sdk_version': mergedProperties['xcode']!,
```
Comment (CaseyHillers) [inline app_dart/lib/src/model/ci_yaml/target.dart:75]: TODO(chillers): Write a regression test

**Comment by CaseyHillers** [comment]
\cc @jmagman

**Review by jmagman — APPROVED** [review]
Code LGTM, @keyonghan should review as well.

File: app_dart/test/model/ci_yaml/target_test.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        'bringup': false,
+        'dependencies': <String>[],
+        '\$flutter/devicelab_osx_sdk': <String, Object>{
+          'sdk_version': '12abc',
+        },
+        'xcode': '12abc',
+      });
+    });
+
+    test('platform properties with xcode', () {
```
Comment (keyonghan) [inline app_dart/test/model/ci_yaml/target_test.dart:38]: We need to have a test when both platform and target properties exist, which is the main purpose of this PR.

Comment (CaseyHillers) [reply app_dart/test/model/ci_yaml/target_test.dart:42]: Added!

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**